Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli

```diff
@@ -0,0 +1,633 @@
+            float3(0.0, 0.0, 0.0) :
+            TransmissionKernel(surface.transmission.thickness * transmissionParams.w, rcp(transmissionParams.xyz)) *
+            saturate(dot(-surface.normal, dirToLight)) * lightIntensity * shadowRatio;
+
+        return result;
+    }
+    else // if ((o_hairLightingModel == HairLightingModel::Marschner) || (o_hairLightingModel == HairLightingModel::Kajiya))
+    {
+        return float3(0.0f, 0.0f, 0.0f);
+    }
+
+    return float3(1.0f, 0.0f, 0.0f);
+}
+
+//! Simple Lambertian BRDF
+float3 HairDiffuseLambertian(float3 albedo, float3 normal, float3 dirToLight)
+{
+    float NdotL = saturate(dot(normal, dirToLight));
+    return albedo * NdotL * INV_PI;
+} 
+
+// Replacing the generic Diffuse and Specular methods in StandardLighting.azsli 
+// and removing the regular usage of clear coat second lobe energy distribution.
+float3 GetDiffuseLighting(Surface surface, LightingData lightingData, float3 lightIntensity, float3 dirToLight)
+{
+    float3 diffuse = float3(0, 1, 0);   // Green - error color
+
+    if (o_hairLightingModel == HairLightingModel::GGX)
+    {
+        // Notice that addition of the response (1-F) here
+        diffuse = HairDiffuseLambertian(surface.albedo, surface.normal, dirToLight) * lightingData.diffuseResponse;
+    }
+    else if (o_hairLightingModel == HairLightingModel::Marschner)
+    {
+        return float3(0.0f, 0.0f, 0.0f);
+    }
+
+    diffuse *= lightIntensity;   
+    return diffuse;
+}
+
+void UpdateLightingParameters(
+    inout LightingData lightingData, 
+    float3 positionWS, float3 normal, float roughnessLinear)
+{
+    lightingData.dirToCamera = normalize(ViewSrg::m_worldPosition.xyz - positionWS);
+
+    // sample BRDF map (indexed by smoothness values rather than roughness)
+    lightingData.NdotV = saturate(dot(normal, lightingData.dirToCamera));
+
+    float2 brdfUV = float2(lightingData.NdotV, (1.0f - roughnessLinear));
+
+    lightingData.brdf = PassSrg::m_brdfMap.Sample(PassSrg::LinearSampler, brdfUV).rg;
+}
+
+//==============================================================================
+void SetNormal(
+    in float3 tangent, in float3 dirToLight, 
+    inout Surface surface, 
+    inout LightingData lightingData)
+{
+    // Direct towards close to half vector by never cross fully (more weight to camera direction)
+    float3 biNormal;
+    if (o_hairLightingModel == HairLightingModel::GGX)
+    {
+        float3 halfDir = normalize( dirToLight + 1.2 * lightingData.dirToCamera);
+        biNormal = normalize(cross(tangent, halfDir));
+    }
+    else
+    {
+        biNormal = normalize(cross(tangent, lightingData.dirToCamera));   // Original - simple face forward
+    }
+
+    float3 projectedNormal = cross(biNormal, tangent);
+    surface.normal = normalize(projectedNormal);    // the normalization might be redundunt 
+
+    // Next is important in order to set NdotV and other PBR settings - needs ot be set once per light
```

Comment:
Typo: "needs **to** be set once per light"

Reply:
tnx